Fix wxDC and wxGCDC clippings in wxQt

AliKet opened the pull request:
With this PR all clipping tests are passing successfully now, which wasn't the case before!

Commit:
Fix wxRegion initialized with negative width or height in wxQt

Commit:
Fix wxDC::Clear() in wxQt

Commit:
Fix applying logical origin for wxDC in wxQt

This makes wxQt works the same as the other ports.

Commit:
Fix wxDC::GetClippingBox() in wxQt

Commit:
Reimplement DoSetClippingRegion() and DoSetDeviceClippingRegion()

For one hand this is a simplification. In the other hand, SetClippingRegion()
and SetDeviceClippingRegion() are both working the same as in the other ports.

Commit:
Simplify setting the clipping region in QtPreparePainter()

Commit:
Make wxGCDC clipping works correctly in wxQt

vadz commented on src/common/dcgraph.cpp:
```diff
@@ -452,7 +452,7 @@ void wxGCDCImpl::DoSetDeviceClippingRegion( const wxRegion &region )
 {
     m_graphicContext->ResetClip();
-#if !defined(__WXOSX__) && !defined(__WXMSW__) && !defined(__WXGTK__)
+#if !defined(__WXOSX__) && !defined(__WXMSW__) && !defined(__WXGTK__) && !defined(__WXQT__)
```
Do we really need this code? AFAICS the only still used port not tested here is wxX11 but it has so many other problems that I'm not sure it's worth to cater to it specifically here.

AliKet replied:
Done.

vadz reviewed (commented):
Very nice, thanks! I can merge this already, but if we can remove the now almost never used code in `wxGCDCImpl::DestroyClippingRegion()`, let's do it too.

Thanks again!

Commit:
Remove no longer used code from wxGCDCImpl::DestroyClippingRegion()

None of the major ports need the removed code anyhow. We can always restore it
later if any of the (still) supported ports are affected by this removal.
